tests/test_components.py

```diff
@@ -0,0 +1,233 @@
+import argparse
+import collections
+import os
+import re
+
+import ssg.build_yaml
+import ssg.components
+import ssg.environment
+import ssg.rules
+import ssg.yaml
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="Test components data consistency")
+    parser.add_argument("--source-dir", help="Path to the root directory")
+    parser.add_argument("--build-dir", help="Path to the build directory")
+    parser.add_argument("--product", help="Product ID")
+    return parser.parse_args()
+
+
+def get_component_by_template(
+        rule, package_to_component, template_to_component):
+    template = rule.template
+    component = None
+    reason = None
+    if not template:
+        return (component, reason)
+    template_name = template["name"]
+    template_vars = template["vars"]
+    if template_name in template_to_component:
+        component = template_to_component[template_name]
+        reason = (
+            "all rules using template '%s' should be assigned to component "
```

Comment:
```suggestion
            "all rules using template '%s' must be assigned to component "
```
I think that if this this behavior cannot be overridden using the world "must" or "shall" would be better. This applies to all the reasons in this file.